soh/soh/ResourceManagerHelpers.cpp

```diff
@@ -503,3 +512,139 @@ extern "C" void ResourceMgr_ClearSkeletons() {
+                auto setRoomListCmd = std::dynamic_pointer_cast<SOH::SetRoomList>(sceneCmd);
+                for (auto room : setRoomListCmd->rooms) {
+                    auto roomScene = (SOH::Scene*)Ship::Context::GetInstance()->GetResourceManager()->LoadResource(room.fileName).get();
+                    for (auto roomSceneCmd : roomScene->commands) {
+                        if (roomSceneCmd->cmdId == SOH::SceneCommandID::SetObjectList) {
+                            auto setObjectCmd = std::dynamic_pointer_cast<SOH::SetObjectList>(roomSceneCmd);
+                            for (auto objectId : setObjectCmd->objects) {
+                                std::string objectName = gObjectTable[objectId].fileName;
+                                sceneObjects[sceneNum].insert(objectName);
+                            }
+                        }
+                        else if (roomSceneCmd->cmdId == SOH::SceneCommandID::SetActorList) {
+                            auto setActorCmd = std::dynamic_pointer_cast<SOH::SetActorList>(roomSceneCmd);
+                            if (setActorCmd->numActors > 0) {
+                                for (uint16_t i = 0; i < setActorCmd->numActors; i++) {
+                                    auto actorEntry = (ActorEntry*)setActorCmd->GetRawPointer();
+                                    std::string objectName = gObjectTable[ActorDB::Instance->RetrieveEntry(actorEntry->id).entry.objectId].fileName;
+                                    sceneObjects[sceneNum].insert(objectName);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    rmhThreadPool->submit_task(std::bind(LoadSceneResourcesProcess, sceneNum));
+}
+
+extern "C" void ResourceMgr_RegisterUnloadSceneAssets(s16 prevScene) {
+    unloadScene = prevScene;
+}
+
+void UnloadSceneAssetsProcess() {
+    for (auto objectName : sceneObjects[unloadScene]) {
+        if (!sceneObjects[gPlayState->sceneNum].contains(objectName)) {
+            std::string objectPath = fmt::format("alt/objects/{}/*", objectName);
+            OTRGlobals::Instance->context->GetResourceManager()->UnloadDirectory(objectPath);
+        }
+    }
+    OTRGlobals::Instance->context->GetResourceManager()->UnloadDirectory(GetScenePath(unloadScene));
+    unloadScene = -1;
+}
+
+extern "C" void ResourceMgr_UnloadSceneAssets() {
+    if (unloadScene != -1) {
+        rmhThreadPool->submit_task(UnloadSceneAssetsProcess);
+    }
+}
+
+extern "C" void ResourceMgr_LoadDelayedPersistentAltAssets() {
+    ResourceLoadDirectoryAsync("alt/textures/vr_cloud*");
+    ResourceLoadDirectoryAsync("alt/overlays/*");
+    ResourceLoadDirectoryAsync("alt/textures/*");
+    ResourceLoadDirectoryAsync("alt/objects/gameplay_*");
+    ResourceLoadDirectoryAsync("alt/scenes/*/spot00*");
+    ResourceLoadDirectoryAsync("alt/code/*");
+}
+
+extern "C" void ResourceMgr_LoadPersistentAltAssets() {
```

Comment:
so is the idea here that whatever someone resets back to will never be unloaded?

Reply:
Nothing in the startup gets unloaded, except maybe spot00, I just don't know. The skyboxes take up the most memory by far, and that's the main common thing between pre-game and in-game. Everything else is mostly incidental to each other.

Reply:
Oh, I get what you're asking. To clarify what's going on here, the Fast File Select options determine the initial load priority for some of the persistent assets, but they all get persisted eventually with the Delayed load.